java/cuvs-java/src/main/java22/com/nvidia/cuvs/internal/BruteForceIndexImpl.java

```diff
@@ -130,8 +126,13 @@ private void checkNotDestroyed() {
   public void destroyIndex() throws Throwable {
     checkNotDestroyed();
     try {
-      int returnValue = cuvsBruteForceIndexDestroy(bruteForceIndexReference.getMemorySegment());
+      int returnValue = cuvsBruteForceIndexDestroy(bruteForceIndexReference.indexPtr);
       checkCuVSError(returnValue, "cuvsBruteForceIndexDestroy");
+
+      if (bruteForceIndexReference.datasetBytes > 0) {
+        returnValue = cuvsRMMFree(resources.getHandle(), bruteForceIndexReference.datasetPtr, bruteForceIndexReference.datasetBytes);
+        checkCuVSError(returnValue, "cuvsRMMFree");
```

Comment:
Yikes.  Am I reading this right?  Was this a memory leak that you just fixed?

Reply:
Ah, I might be mistaken.  The `IndexReference` now has two `MemorySegment`s:  One for the reference pointer, and another for the actual data.

Reply:
Yeah, I think there was no `cuvsRMMFree` for the data; the `IndexReference` now has two `MemorySegment`s because we need to track the "dataset pointer" too in order to free it when we are done.